test/test.js

```diff
@@ -743,6 +743,17 @@ describe('Page', function() {
       process.removeListener('warning', warningHandler);
       expect(warning).toBe(null);
     }));
+    it('should remove event listeners when page is closed', SX(async function() {
+      const closedPage = await browser.newPage();
+      await closedPage.close();
+
+      process.once('warning', warning => expect(warning).toBe(null));
```

Comment:
this event listener should be removed after the test.

note: You can just copy the previous test and use 'asdf' instead of EMPTY_PAGE as url:

```js
    it('should not leak listeners during bad navigation', SX(async function() {
      let warning = null;
      const warningHandler = w => warning = w;
      process.on('warning', warningHandler);
      for (let i = 0; i < 20; ++i)
        await page.goto('asdf').catch(e => {/* swallow navigation error */});
      process.removeListener('warning', warningHandler);
      expect(warning).toBe(null);
    }));
```

Reply:
Oh, I fixed it.

Thank you 😃